document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java

```diff
@@ -79,56 +76,64 @@ public DocumentInfoResponse getAbridgedAccountsData(String resourceLink, String
+            return accounts.getLinks().getTransaction();
+        } else {
+            throw new  HandlerException("Failed to get transaction link for resource Uri: "
+                    + resourceUri);
+        }
+    }
+
+    private Transaction getTransaction(String transactionLink, String requestId)
+            throws HandlerException {
+
+        try {
+            return transactionService.getTransaction(transactionLink, requestId);
+        } catch (ServiceException e) {
             Map<String, Object> logMap = new HashMap<>();
-            logMap.put(RESOURCE, abridgedAccountLink);
-            logMap.put(ACCOUNT_TYPE, accountType);
-            LOG.errorContext(requestId,"Error when parsing period end on date from smallFull accounts data", e, logMap);
-            throw new HandlerException(e.getMessage(), e.getCause());
+            logMap.put(TRANSACTION_LINK, transactionLink);
+            LOG.errorContext(requestId,"An error occurred when calling the transaction service with transaction link: "
+                    + transactionLink, e, logMap);
+            throw new HandlerException("Failed to get transaction with transaction link: " + transactionLink, e);
         }
     }
 
     private Accounts getAccounts(String resourceLink, String requestId) throws HandlerException {
 
-        Accounts accounts;
-
         try {
-            accounts = accountsService.getAccounts(resourceLink, requestId);
+             return accountsService.getAccounts(resourceLink, requestId);
         } catch (ServiceException e) {
             Map<String, Object> logMap = new HashMap<>();
             logMap.put(RESOURCE, resourceLink);
             LOG.errorContext(requestId,"Error in service layer when obtaining accounts data for resource: "
                     + resourceLink, e, logMap);
             throw new HandlerException(e.getMessage(), e.getCause());
         }
-
-        return accounts;
     }
 
-    /**
-     * Get the account type from the links resource within the given accounts data object
-     *
-     * @param accounts accounts resource data
-     * @return the {@link AccountType} that exist in the given accounts data
-     * @throws HandlerException if unable to find account type in accounts data
-     */
     private AccountType getAccountType(Accounts accounts) throws HandlerException {
-        return accounts.getLinks().keySet()
-                .stream()
-                .filter(e -> !e.equalsIgnoreCase(LinkType.SELF.getLink()))
-                .map(AccountType::getAccountType)
-                .findFirst()
-                .orElseThrow(() -> new HandlerException("Unable to find account type in account data" +
-                        accounts.getId()));
+
+        if (accounts.getLinks().getAbridgedAccounts() != null) {
+            return AccountType.getAccountType("abridged_accounts");
+        } else {
+            throw new HandlerException("Unable to find account type in account data" +
```

Comment:
There needs to be a space after the word 'data' here, like `... account data " + ...`. Plus, should we use something more distinct than the Accounts `kind` for the logs? Do we have access to a unique identifier?